vg/vg.go

```diff
@@ -85,6 +88,49 @@ type CanvasSizer interface {
 	Size() (x, y Length)
 }
 
+// CanvasWriterTo wraps behavior required for writing
+// a Canvas representation to a concrete image.
+type CanvasWriterTo interface {
+	CanvasSizer
+	io.WriterTo
+}
+
+var (
+	formatLock sync.RWMutex
+	formats    = make(map[string]func(w, h Length) CanvasWriterTo)
+)
+
+// Register allows a format to be registered with the vg package.
+// Registered formats can be returned by NewCanvasWriterTo.
+func Register(format string, new func(w, h Length) CanvasWriterTo) {
+	formatLock.Lock()
+	formats[format] = new
```

Comment:
OK, I'll do that. I also need to add to plot Save that the extension switch is case insensitive.